plugins/clnrest/clnrest.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+# For --hidden-import gunicorn.glogging gunicorn.workers.sync
+from gunicorn import glogging
+from gunicorn.workers import sync
+
+from pathlib import Path
+from flask import Flask
+from flask_restx import Api
+from gunicorn.app.base import BaseApplication
+from multiprocessing import Process, cpu_count
+from utilities.generate_certs import generate_certs
+from utilities.shared import set_config
+from utilities.rpc_routes import rpcns
+from utilities.rpc_plugin import plugin
+
+jobs = {}
+
+def create_app():
+    app = Flask(__name__)
+    authorizations = {
+        "rune": {"type": "apiKey","in": "header","name": "Rune"},
+        "nodeid": {"type": "apiKey","in": "header","name": "Nodeid"}
+    }
+    api = Api(app, version="1.0", title="Core Lightning Rest", description="Core Lightning REST API Swagger", authorizations=authorizations, security=["rune", "nodeid"])
+    api.add_namespace(rpcns, path="/v1")
+    return app
+
+def set_application_options(plugin):
+    from utilities.shared import CERTS_PATH, REST_PROTOCOL, REST_HOST, REST_PORT
+    plugin.log(f"REST Server is starting at {REST_PROTOCOL}://{REST_HOST}:{REST_PORT}", "debug")
+    if REST_PROTOCOL == "http":
+        options = {
+            "bind": f"{REST_HOST}:{REST_PORT}",
+            "workers": cpu_count(),
+            "timeout": 60,
+            "loglevel": "warning",
+        }
+    else:
+        cert_file = Path(f"{CERTS_PATH}/client.pem")
+        key_file = Path(f"{CERTS_PATH}/client-key.pem")
+        if not cert_file.is_file() or not key_file.is_file():
+            plugin.log(f"Certificate not found at {CERTS_PATH}. Generating a new certificate!", "debug")
+            generate_certs(plugin, CERTS_PATH)
+        try:
+            plugin.log(f"Certs Path: {CERTS_PATH}", "debug")
+        except Exception as err:
+            raise Exception(f"{err}: Certificates do not exist at {CERTS_PATH}")
+        options = {
+            "bind": f"{REST_HOST}:{REST_PORT}",
+            "workers": cpu_count(),
+            "timeout": 60,
+            "loglevel": "warning",
+            "certfile": f"{CERTS_PATH}/client.pem",
+            "keyfile": f"{CERTS_PATH}/client-key.pem",
+        }
+    return options
+
+class CLNRestApplication(BaseApplication):
+    def __init__(self, app, options=None):
+        from utilities.shared import REST_PROTOCOL, REST_HOST, REST_PORT
+        self.application = app
+        self.options = options or {}
+        plugin.log(f"REST server running at {REST_PROTOCOL}://{REST_HOST}:{REST_PORT}", "info")
+        super().__init__()
```

Comment:
`super().__init__()` as the convention is the first call that is done inside the constructor

Reply:
Adding this call first throws an error as "Error: 'CLNRestApplication' object has no attribute 'options'" and does not start the plugin at all. So I had to add options and application before I could initialise BaseApplication class.